lib/cache/src/filesystem.rs

```diff
@@ -63,8 +63,15 @@ impl FileSystemCache {
             }
         } else {
             // Create the directory and any parent directories if they don't yet exist.
-            create_dir_all(&path)?;
-            Ok(Self { path, ext: None })
+            let res = create_dir_all(&path);
+            if res.is_err() {
+                Err(io::Error::new(
+                    io::ErrorKind::Other,
```

Comment:
Why not using `io::ErrorKind::PermissionDenied` here? (cf https://doc.rust-lang.org/std/io/enum.ErrorKind.html)

Reply:
I was also hesitant to choose between `PermissionDenied` and `Other`, the reason I choose `Other` is considering we have some other reasons for failing to create a directory ....

`PermissionDenied` is also OK, since I mainly want to print a detailed message when the cache directory is failed to create.

Reply:
Good to me.